cmd/list/instances.go

```diff
@@ -81,5 +84,31 @@ func executeListInstancesCmd(cmd *cobra.Command, args []string, opts *InstancesO
 	configAndStacksInfo.Command = "list"
 	configAndStacksInfo.SubCommand = "instances"
 
-	return list.ExecuteListInstancesCmd(&configAndStacksInfo, cmd, args)
+	// Load atmos configuration to get auth config.
+	atmosConfig, err := cfg.InitCliConfig(schema.ConfigAndStacksInfo{}, false)
+	if err != nil {
+		return err
+	}
+
+	// Get identity from --identity flag (inherited from parent list command) or ATMOS_IDENTITY env var.
+	// Precedence: flag > env var > stack-level default (scanned by CreateAndAuthenticateManagerWithAtmosConfig).
+	identityName := ""
+	if cmd.Flags().Changed("identity") {
+		identityName, _ = cmd.Flags().GetString("identity")
+	} else if envIdentity := viper.GetString("identity"); envIdentity != "" {
+		identityName = envIdentity
+	}
```

Comment:
I believe we have a function for this; we should avoid reimplementing it, so we have a single code path.